static/js/protein/data_processing_utils.ts

```diff
@@ -0,0 +1,400 @@
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let score = null;
+      let variant = null;
+      let tissue = null;
+      let interval = null;
+      for (const n of node.neighbors) {
+        if (n.property === "geneSymbol") {
+          for (const n1 of n.nodes) {
+            for (const n2 of n1.neighbors) {
+              if (n2.property === "referenceSNPClusterID") {
+                if (n2.nodes[0].value !== null) {
+                  variant = n2.nodes[0].value;
+                }
+              } else if (n2.property === "log2AllelicFoldChange") {
+                if (n2.nodes[0].value !== null) {
+                  score = n2.nodes[0].value;
+                }
+              } else if (
+                n2.property === "log2AllelicFoldChangeConfidenceInterval"
+              ) {
+                if (n2.nodes[0].value !== null) {
+                  interval = n2.nodes[0].value;
+                }
+              } else if (n2.property === "associatedTissue") {
+                if (n2.nodes[0].value) {
+                  tissue = n2.nodes[0].value;
+                }
+              }
+            }
+            if (!seen.has(variant) && !!score) {
+              returnData.push({
+                id: variant,
+                name: tissue,
+                value: score,
+                interval: interval,
+              });
+            }
+            seen.add(variant);
+          }
+        }
+      }
+    }
+    return returnData;
+  }
+  return [];
+}
+
+export function getVarTypeAssoc(data: GraphNodes): ProteinNumData[] {
+  // Variant Gene Associations
+  if (!data) {
+    return [];
+  }
+  const result: ProteinNumData[] = [];
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "geneID") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let variant = null;
+      for (const n of node.neighbors) {
+        if (n.property !== "geneID") {
+          continue;
+        }
+        for (const n1 of n.nodes) {
+          for (const n2 of n1.neighbors) {
+            if (n2.property === "geneticVariantFunctionalCategory") {
+              //check for empty values
+              if (_.isEmpty(n2.nodes[0].value)) {
+                continue;
+              }
+              const count = 1;
+              variant = n2.nodes[0].value;
+              result.push({ name: variant, value: count });
+            }
+          }
+        }
+      }
+    }
+    const returnData = [];
+    // combine all the variant counts based on similar categories
+    result.forEach(function (element) {
+      if (!this[element.name]) {
+        this[element.name] = { name: element.name, value: 0 };
+        returnData.push(this[element.name]);
+      }
+      this[element.name].value += element.value;
+    }, Object.create(null));
+    return returnData;
```

Comment:
Please take a look at my comment about this function from the first round of review on this PR

Reply:
Removed the forEach statement and replaced it with maps!